scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/SFTPConnector.java

```diff
@@ -0,0 +1,378 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.scheduler.examples;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.Serializable;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import org.ow2.proactive.scheduler.common.task.TaskResult;
+import org.ow2.proactive.scheduler.common.task.executable.JavaExecutable;
+import org.zeroturnaround.zip.ZipUtil;
+
+import com.jcraft.jsch.Channel;
+import com.jcraft.jsch.ChannelSftp;
+import com.jcraft.jsch.JSch;
+import com.jcraft.jsch.JSchException;
+import com.jcraft.jsch.Session;
+import com.jcraft.jsch.SftpException;
+
+
+/**
+ * Import and export file(s) using SFTPConnector server.
+ *
+ * @author The ProActive Team
+ */
+public class SFTPConnector extends JavaExecutable {
+
+    private static final String DEFAULT_SFTP_HOSTNAME = "localhost";
+
+    private static final int DEFAULT_SFTP_PORT = 22;
+
+    private String sftpHostname = DEFAULT_SFTP_HOSTNAME;
+
+    private int sftpPort = DEFAULT_SFTP_PORT;
+
+    private boolean sftpExtractArchive;
+
+    private String sftpLocalRelativePath;
+
+    private String sftpRemoteRelativePath;
+
+    private String sftpMode;
+
+    private String sftpUsername;
+
+    private String sftpUrlKey;
+
+    private String sftpPassword;
+
+    private static final String SFTP_LOCAL_RELATIVE_PATH_ARG = "sftpLocalRelativePath";
+
+    @Override
+    public void init(Map<String, Serializable> args) throws Exception {
+        if (args.containsKey("sftpHostname")) {
+            sftpHostname = args.get("sftpHostname").toString();
+        }
+        if (args.containsKey("sftpPort")) {
+            sftpPort = Integer.parseInt(args.get("sftpPort").toString());
+        }
+        if (args.containsKey("sftpMode")) {
+            sftpMode = args.get("sftpMode").toString();
+        }
+        if (args.containsKey("sftpExtractArchive")) {
+            sftpExtractArchive = Boolean.parseBoolean(args.get("sftpExtractArchive").toString());
+        }
+        if (args.containsKey(SFTP_LOCAL_RELATIVE_PATH_ARG) &&
+            !args.get(SFTP_LOCAL_RELATIVE_PATH_ARG).toString().isEmpty()) {
+            sftpLocalRelativePath = args.get(SFTP_LOCAL_RELATIVE_PATH_ARG).toString();
+        } else {
+            //we throw an exception tp prevent transferring all the contents of the global space.
+            if (sftpMode.equals("PUT")) {
```

Comment:
Can you create a constant for the "PUT" String?

Reply:
Agree